src/wallet/test/db_tests.cpp

```diff
@@ -228,6 +228,52 @@ BOOST_AUTO_TEST_CASE(db_availability_after_write_error)
     }
 }
 
+#ifdef USE_SQLITE
+
+class DbExecBlocker : public SQliteExecHandler
+{
+public:
+    bool CanExecute(const std::string& statement) override {
+        if (statement == "ROLLBACK TRANSACTION") return false;
+        return true;
+    }
+};
+
+BOOST_AUTO_TEST_CASE(txn_close_failure_dangling_txn)
+{
+    // Verifies that there is no active dangling, to-be-reversed db txn
+    // after the batch object that initiated it is destroyed.
+    DatabaseOptions options;
+    DatabaseStatus status;
+    bilingual_str error;
+    std::unique_ptr<SQLiteDatabase> database = MakeSQLiteDatabase(m_path_root / "sqlite", options, status, error);
+
+    std::string key = "key";
+    std::string value = "value";
+
+    std::unique_ptr<SQLiteBatch> handler = std::make_unique<SQLiteBatch>(*database);
```

Comment:
In commit "test: sqlite, add coverage for dangling to-be-reverted db txns" (d4ebac8381801d024b69d5f1f5a13246c61edbe4)

I think it would be a little more consistent with other code to call this variable "batch" instead of "handler." Also calling it handler gets a little confusing when code below sets a handler on the handler.

Reply:
So true that it hurts.
I tend to call the `WalletBatch` handler because 'batch' refers to a class that groups operations and dumps the result all at once atomically at the end of the process. This is something that the `WalletBatch` class doesn't do unless we explicitly begin and finish the transaction.